src/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -289,6 +289,65 @@ contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistr
+     * has their StakeRegistry.quorumTimestamp updated. which is meant to keep track of when operators were last all updated at once. 
+     * @param operatorsPerQuorum is an array of arrays of operators to update for each quorum. Note that each nested array
+     * of operators must be sorted in ascending operatorId order to ensure that all operators in the quorum are updated
+     * @param quorumNumbers is an array of quorum numbers to update
+     * @dev This method is used to update the stakes of all operators in a quorum at once, rather than individually
+     * 
+     */
+    function updateOperatorsForQuorum(address[][] calldata operatorsPerQuorum, bytes calldata quorumNumbers) external onlyWhenNotPaused(PAUSED_UPDATE_OPERATOR) {
+        require(
+            operatorsPerQuorum.length == quorumNumbers.length,
+            "BLSRegistryCoordinatorWithIndices.updateOperatorsForQuorum: input length mismatch"
+        );
+        for (uint256 i = 0; i < quorumNumbers.length; ++i) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            address[] memory currQuorumOperators = operatorsPerQuorum[i];
+            require(
+                currQuorumOperators.length == indexRegistry.totalOperatorsForQuorum(quorumNumber),
+                "BLSRegistryCoordinatorWithIndices.updateOperatorsForQuorum: number of updated operators does not match quorum total"
+            );
+            address prevOperatorAddress = address(0);
+            // Update stakes for each operator in this quorum
+            for (uint256 j = 0; j < currQuorumOperators.length; ++j) {
+                address operator = currQuorumOperators[j];
+                bytes32 operatorId = getOperatorId(operator);
+                uint192 currentBitmap = _currentOperatorBitmap(operatorId);
+
+                /**
+                 * Assuming the operatorId cannot be 0, for the first operatorId, prevOperatorId will be 0
+                 * Require check is to prevent duplicate operators and that all quorum operators are updated
+                 */
+                require(
+                    operator > prevOperatorAddress,
+                    "BLSRegistryCoordinatorWithIndices.updateOperatorsForQuorum: operators array must be sorted in ascending operatorId order"
+                );
+                require(
+                    BitmapUtils.numberIsInBitmap(currentBitmap, quorumNumber),
+                    "BLSRegistryCoordinatorWithIndices.updateOperatorsForQuorum: operator not in quorum"
+                );
+
+                /**
+                 * Update the operator's stake for their active quorums. The stakeRegistry returns a bitmap
+                 * of quorums where the operator no longer meets the minimum stake, and should be deregistered.
+                 */
+                uint192 quorumsToRemove = stakeRegistry.updateOperatorStake(operator, operatorId, BitmapUtils.bitmapToBytesArray(1<<quorumNumber));                
```

Comment:
The call to `BitmapUtils.bitmapToBytesArray` is ~fairly gas-expensive here, as it explicitly does 256 loop iterations.
This function should only be used in general when it's needed.
We should be able to do a byte slice here instead, like `quorumNumbers[i:i+1]` -- I can't remember where, but we are already using this in at least one other middleware contract.

Reply:
It’s this contract. Good call

Reply:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/62/commits/0556f9276126cebd4eecde019282670365a175db Updated with bytes slicing here as well as for the _deregisterOperator call since quorumsToRemove will only be 0 or with the specific quorum bit toggled